src/database.rs

```diff
@@ -0,0 +1,186 @@
+use async_trait::async_trait;
+use std::str::FromStr;
+
+use log::{info, trace, warn};
+use sqlx::sqlite::{SqliteConnectOptions, SqlitePoolOptions};
+
+use crate::{types::AstarteType, AstarteError, AstarteSdk};
+
+#[derive(Clone)]
+pub struct Database {
+    db_conn: sqlx::Pool<sqlx::Sqlite>,
+}
+
+#[async_trait]
+pub trait AstarteDatabase {
+    async fn store_prop(&self, key: &str, value: &[u8], version: i32) -> Result<(), AstarteError>;
+    async fn load_prop(&self, key: &str, version: i32)
+        -> Result<Option<AstarteType>, AstarteError>;
+    async fn delete_prop(&self, key: &str) -> Result<(), AstarteError>;
+
+    async fn clear(&self) -> Result<(), AstarteError>;
+}
+
+#[async_trait]
+impl AstarteDatabase for Database {
+    async fn store_prop(&self, key: &str, value: &[u8], version: i32) -> Result<(), AstarteError> {
```

Comment:
Use `interface_major` instead of `version` both here and in the column, otherwise it's not clear what the version is referring to